src/components/Avatar/Avatar.module.scss

```diff
@@ -0,0 +1,180 @@
+@import '../../styles/colors';
+
+.Avatar {
+  &:focus-within {
+    --zui-avatar-border-radius: 50%;
+  }
+
+  .AvatarRoot {
+    --zui-avatar-border-radius: 50%;
+    --zui-avatar-width: 2.5rem;
+    --zui-avatar-height: 2.5rem;
+  }
+
+  &[data-type='square'] {
+    .AvatarRoot {
+      --zui-avatar-border-radius: 0.5rem;
+    }
+
+    &:focus-within {
+      --zui-avatar-border-radius: 0.5rem;
+    }
+  }
+
+  .AvatarFallback {
+    --zui-avatar-font-size: 0.875rem;
+    --zui-avatar-font-line-height: 150%;
+  }
+
+  .Badge {
+    --zui-avatar-badge-position-right: -0.375rem;
+    --zui-avatar-badge-position-top: -0.375rem;
+  }
+
+  &[data-size='small'] {
+    .AvatarRoot {
+      --zui-avatar-width: 1.5rem;
+      --zui-avatar-height: 1.5rem;
+    }
+
+    .Badge {
+      --zui-avatar-badge-position-top: -0.625rem;
+      --zui-avatar-badge-position-right: -0.625rem;
+    }
+
+    .AvatarFallback {
+      --zui-avatar-font-size: 0.625rem;
+      --zui-avatar-font-line-height: 0.75rem;
+    }
+  }
+
+  &[data-size='extra small'] {
+    .AvatarRoot {
+      --zui-avatar-width: 1rem;
+      --zui-avatar-height: 1rem;
+    }
+
+    .AvatarFallback {
+      --zui-avatar-font-size: 0.5rem;
+      --zui-avatar-font-line-height: 0.625rem;
+    }
+  }
+
+  &[data-size='medium'] {
+    .AvatarRoot {
+      --zui-avatar-width: 2rem;
+      --zui-avatar-height: 2rem;
+    }
+
+    .Badge {
+      --zui-avatar-badge-position-top: -0.625rem;
+      --zui-avatar-badge-position-right: -0.625rem;
+    }
+
+    .AvatarFallback {
+      --zui-avatar-font-size: 0.625rem;
+      --zui-avatar-font-line-height: 0.75rem;
+    }
+  }
+
+  .Status {
+    &[data-status-type='active'] {
+      --zui-avatar-status-background: #{$green-9};
+    }
+
+    &[data-status-type='idle'] {
+      --zui-avatar-status-background: #{$orange-10};
+    }
+
+    &[data-status-type='busy'] {
+      --zui-avatar-status-background: #{$read-10};
+    }
+
+    &[data-status-type='offline'] {
+      --zui-avatar-status-background: #{$grey-10};
+    }
+
+    &[data-status-type='unread'] {
+      --zui-avatar-status-background: #{$purple-11};
+    }
+  }
+}
+
+.Avatar {
+  position: relative;
+  display: flex;
+  cursor: pointer;
+
+  &:focus-within {
+    border-radius: var(--zui-avatar-border-radius);
+    outline-offset: 0.25rem;
+    outline: 0.0625rem solid $green-11;
+  }
+
+  .AvatarRoot {
```

Comment:
You can remove `Avatar` from these classes - the module will make sure they're scoped properly.

Just `.Avatar {  .Root { } }` is fine!

Reply:
Updated